version: '3'
services:
  plex:
    image: linuxserver/plex:latest
    container_name: plex
    restart: unless-stopped
    environment:
      - PUID=${PUID}
      - PGID=${PGID}
      - TZ=America/Los_Angeles
      - ADVERTISE_IP=plex.kevinc.cloud:443
    networks:
      - traefik
    ports:
      - 32400:32400/tcp
      - 8324:8324/tcp
      - 32469:32469/tcp
      - 1900:1900/udp
      - 32410:32410/udp
      - 32412:32412/udp
      - 32413:32413/udp
      - 32414:32414/udp

    volumes:
      - /volume1/docker/plex:/config:rw
      - /volume1/Data:/Data:rw
    labels:
      my.zone: homeserver
      traefik.enable: true
      traefik.http.routers.plex.rule: Host(`plex.${SERVER_DOMAIN}`)
      traefik.http.routers.plex.entrypoints: websecure
      traefik.http.services.plex.loadbalancer.server.port: 32400

networks:
  traefik:
    name: homeserver
    external: true